### DISCUSSION TITLE
Binding SPC SPC to a command

### DISCUSSION THREAD

**Discussion by zetashift** [meow-edit/meow#482] [Q&A] [answered]
I'm wondering how to bind `consult-find-file` to `SPC SPC` in meow, when in normal mode.

I'd tried to add a `<space>` entry in `meow-leader-define-key` but nothing happened.

**Comment by DogLooksGood** [comment]
`SPC` in normal mode brings you to the keypad mode. And in keypad mode, `SPC` is used as literal prefix. So there's no easy way to bind `SPC SPC`. You have to add advice to `meow-keypad-self-insert` command, and check if it's the first `SPC` in keypad mode.

**Reply by zetashift** [reply]
Ah damn, my emacs-lisp-fu is way too weak for that, I'll try okamsn route

**Comment by okamsn** [comment]
What I did was to set Meow to use different map as the leader map, which seems to be working.

    (setf meow-keypad-leader-dispatch my-leader-map
          (map-elt meow-keymap-alist 'leader) my-leader-map)

**Reply by zetashift** [reply]
I have not been able to get this working because of bad elisp, how exactly is `my-leader-map` defined?

**Comment by okamsn** [comment, marked as answer]
> I have not been able to get this working because of bad elisp, how exactly is `my-leader-map` defined?

It is just a normal keymap :

```emacs-lisp
;; Bind variable to a non-sparse (as in not from `make-sparse-keymap') empty keymap:
(defvar my-leader-map (make-keymap) "My leader map.")

;; Make Meow use the leader key:
(setf meow-keypad-leader-dispatch my-leader-map
      (map-elt meow-keymap-alist 'leader) my-leader-map)

;; Bind key sequence to a command in the keymap contained by the variable:
(define-key my-leader-map (kbd "<SPC>") #'execute-extended-command)

```

If I recall correctly, I originally used `make-sparse-keymap`, but that didn't work for some reason.

In newer Emacs, it looks like you can do

```emacs-lisp
(defvar-keymap my-leader-map2
  :doc "My leader map, again."
  :full t
  "<SPC>" #'execute-extended-command
  ... )
```

It's probably relevant that I don't have any keybindings with modifiers in them in the first layer of my leader map.

**Reply by zetashift** [reply]
Sweet, this worked nicely, thank you!
